### PR TITLE
[MRG+1] Adding requests referer to RFPDupeFilter log messages

### PR DESCRIPTION
**Pull request by ejulio**
I just changed the log message when DUPEFILTER_DEBUG is active because it helps knowing which request is generating the duplicates (I'd this problem recently).

Also, I added some tests for the log method to check the proper messages and stats.

### CODE REVIEW

**Comment by dangra** [comment]
@kmike if you don't see any objections, I'll merge and ship it in 1.6.0

**Comment by dangra** [comment]
thanks Julio.

File: scrapy/dupefilters.py [hunk 1]
```diff
@@ -61,8 +60,9 @@ def close(self, reason):
 
     def log(self, request, spider):
         if self.debug:
-            msg = "Filtered duplicate request: %(request)s"
-            self.logger.debug(msg, {'request': request}, extra={'spider': spider})
+            msg = "Filtered duplicate request: %(request)s (referer: %(referer)s)"
+            args = {'request': request, 'referer': referer_str(request) }
```
Comment (kmike) [inline scrapy/dupefilters.py]: Hey! Sorry for delay. No objections to me, a nice feature. I was just wondering if Referer should be added to the next message as well, when debug is False.

Comment (ejulio) [reply scrapy/dupefilters.py]: Fair point.
I just added to keep the same behavior, but it makes sense to log referer only in debug mode.

**Pull request merged**